baybe/constraints/continuous.py

```diff
@@ -31,3 +35,91 @@ class ContinuousLinearInequalityConstraint(ContinuousConstraint):
     The class has no real content as it only serves the purpose of
     distinguishing the constraints.
     """
+
+
+@define
+class ContinuousCardinalityConstraint(ContinuousConstraint):
+    """Class for continuous cardinality constraints.
+
+    Places a constraint on the set of nonzero (i.e. "active") values among the
+    specified parameters, bounding it between the two given integers,
+        ``min_cardinality`` <= |{p_i : p_i != 0}_i| <= ``max_cardinality``
```

Comment:
```suggestion
        ``min_cardinality`` <= |{p_i : p_i != 0}| <= ``max_cardinality``
```
last `_i` is not necessary right?

Reply:
Hm, depends on how sloppy we want to be with the math :D I think technically it would be more "correct" to explicitly state what is the index (if we want to be totally correct, we'd even need to specify the set over which we index). If we omit it, then it would mean that we expect everyone just silently assumes "i" is the index. Happy with both versions, let me know what you prefer

Reply:
but your indexing the set, is that commonly done? seems strange to me

an index is not needed for that set notation eg `{x: x in (3,10]}` talks about all `x`, the only implicit assumption is x is a real number too. `|{p_i: p_i != 0}|` talks already about all `p_i`

Reply:
Yes, I think having a notation `{...}_{i=1}^N` or `{...}_{i \in S}` is very common. There is a difference between the two cases you describe:
* in the former, the content of the set is explicitly clear from your "in" part. It's all number is the half-open interval
* in the latter, it's unclear what we even iterate over to check the condition. That is, if "i" is the index, what is even the "base set" of `p_i`s? I.e., is the base set `{p_1, p_2}` or `{p_1, p_2, p_3}`? And a pedantic reader could even say that the index could even be "p", i.e. that we iterate over `{firstobject_i, secondobject_i, ...}`.

But I agree that we don't have to overcomplicate things here. Since you seem to prefer the version without the "i", let me simply remove it 🙃

Reply:
curious now, can you show an example where `_` and `^` are used with set builder notation? I know its very common for sums etc, but for sets? talking things like 
![image](https://github.com/emdgroup/baybe/assets/17951239/62643cfb-83b2-4834-8de2-5e5d2fd722eb)

given that we show this notation for a cosntraint acting on a set of parameters I think the base set of `p_i` is rather clear, I also wouldnt object loosing the `i` entirely

Reply:
https://math.stackexchange.com/questions/4033995/set-notation-with-subscript-and-superscript